forms/LiteralField.php

```diff
@@ -1,76 +1,103 @@
 <?php
+
 /**
  * This field lets you put an arbitrary piece of HTML into your forms.
- * 
- * <b>Usage</b>
- * 
+ *
  * <code>
  * new LiteralField (
  *    $name = "literalfield",
  *    $content = '<b>some bold text</b> and <a href="http://silverstripe.com">a link</a>'
  * )
  * </code>
- * 
+ *
  * @package forms
  * @subpackage fields-dataless
  */
 class LiteralField extends DatalessField {
-	
 	/**
-	 * @var string $content
+	 * @var string|FormField
 	 */
 	protected $content;
-	
+
+	/**
+	 * @param string $name
+	 * @param string|FormField $content
+	 */
 	public function __construct($name, $content) {
-		$this->content = $content;
-		
+		$this->setContent($content);
+
 		parent::__construct($name);
 	}
-	
+
+	/**
+	 * @param array $properties
+	 *
+	 * @return string
+	 */
 	public function FieldHolder($properties = array()) {
-		if(is_object($this->content)) {
-			$obj = $this->content;
-			if($properties)
-				$obj = $obj->customise($properties);
-			return $obj->forTemplate();
-		} else {
-			return $this->content;
+		if($this->content instanceof FormField) {
+			$context = $this->content;
```

Comment:
Could the `$this->context` possibly be intended for DBField as well as FormField? Maybe it should be `instanceof ViewableData`? (Although even that method doesn't implement forTemplate, it's the highest common denominator of classes that do).